-sandbox
# BRFSS 2020 DEMOGRAPHIC DATA

- **Dataset**

The Behavioral Risk Factor Surveillance System (BRFSS) is an annual telephone survey conducted by the Centers for Disease Control and Prevention (CDC). The BRFSS collects data on health-related risk behaviors, chronic health conditions, and use of preventive services for adults aged 18 and over in the United States.

The 2020 BRFSS collected data from 463,323 adults aged 18 and over. Demographic information collected included gender, age, race/ethnicity, education level, marital status, employment status, annual household income, and health insurance status.

Data Source - https://www.kaggle.com/datasets/aemreusta/brfss-2020-survey-data

columns - 279

size - 323.288 MB

- STATE: The two-digit code for the state

- STATENAME: The name of the state 

- AGE: The age of the respondent

- SEXVAR: A gender indicator (1=Male, 2=Female)

- MARITAL: The marital status of the respondent (1=Married, 2=Never Married, 3=Divorced, 4=Widowed)

- EDUCA: The highest level of education obtained by the respondent (1=Less than High School, 2=High School/GED, 3=Some College, 4=College Graduate, 5=Post Graduate Degree)

- RENTHOME: Indicator of whether respondent rents or owns their home (1=Rents, 2=Owns)

- CELLPHONES: Indicator of whether respondent has a cell phone (1=Yes, 2=No)

- VETERAN: Indicator of whether respondent is a veteran (1=Yes, 2=No)

- EMPLOYE: Indicator of whether respondent is employed (1=Employed, 2=Unemployed)

- INCOME: The annual household income of the respondent

- CHILDREN: Indicator of whether respondent has children (1=Yes, 2=No)

- WEIGHT: The body weight of the respondent in kilograms

- HEIGHT: The body height of the respondent in centimeters

- BMI: The body mass index of the respondent

- GENHLTH: The self-reported general health of the respondent (1=Excellent, 2=Very Good, 3=Good, 4=Fair, 5=Poor)

- **Importing Libraries**

In [0]:
from pyspark.sql.types import *
from pyspark.sql.functions import *
import pyspark.sql.functions as F
import pandas as pd
from matplotlib import pyplot as plt
%matplotlib inline

- **Loading the dataset**

In [0]:
filePath = "/tmp/sf-brfss2020.parquet"
brfss2020_df = spark.read.parquet(filePath)
display(brfss2020_df)

STATE AGE SEXVAR MARITAL EDUCA RENTHOME CELLPHONES VETERAN EMPLOYE INCOME CHILDREN WEIGHT HEIGHT BMI ASTHMA ARTHRITIS DEPRESSION DIABETE HEARTDISEASE STROKE HIV HEARTATT CONFUSSION SMOKE100 USENOW3 CURRENTSMOKE ALCDAY5 DRUNKDAY DRUNKWEEK DRUNKHEAVY SMOKESTATUS DRINKS SLEPTIME PHYEXERCISE HEALTH PHYHLTH MENTHLTH RACE ECIGARET GENHLTH STATENAME 1.0 8.0 2.0 2.0 6.0 1.0 1.0 2.0 4.0 1.0 88.0 106.0 507.0 1.0 1.0 1.0 1.0 1.0 2.0 2.0 1.0 2.0 null 1.0 3.0 2.0 888.0 2.0 0.0 1.0 1.0 1.0 5.0 1.0 1.0 2.0 3.0 1.0 1.0 2.0 Alabama 1.0 10.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 99.0 88.0 170.0 504.0 3.0 1.0 1.0 1.0 3.0 2.0 2.0 null 2.0 null null null 9.0 null 9.0 99900.0 9.0 9.0 9.0 7.0 1.0 1.0 1.0 1.0 2.0 null 3.0 Alabama 1.0 10.0 2.0 1.0 5.0 1.0 1.0 2.0 7.0 7.0 88.0 7777.0 508.0 null 2.0 1.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 7.0 1.0 1.0 1.0 1.0 2.0 2.0 3.0 Alabama 1.0 13.0 2.0 3.0 4.0 1.0 9.0 2.0 5.0 99.0 88.0 9999.0 9999.0 null 2.0 2.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 6.0 2.0 1.0 1.0 1.0 1.0 2.0 1.0 Alabama 1.0 13.0 2.0 3.0 6.0 2.0 8.0 2.0 7.0 77.0 88.0 126.0 506.0 2.0 2.0 2.0 2.0 3.0 2.0 1.0 9.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 7.0 1.0 1.0 1.0 1.0 1.0 2.0 2.0 Alabama 1.0 10.0 1.0 4.0 4.0 3.0 1.0 2.0 8.0 5.0 88.0 180.0 509.0 3.0 1.0 1.0 2.0 1.0 2.0 2.0 1.0 2.0 null 1.0 1.0 1.0 888.0 2.0 0.0 1.0 3.0 1.0 8.0 1.0 2.0 3.0 3.0 1.0 1.0 4.0 Alabama 1.0 12.0 2.0 1.0 4.0 1.0 2.0 2.0 7.0 6.0 88.0 150.0 506.0 2.0 2.0 2.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 6.0 2.0 1.0 1.0 1.0 1.0 2.0 3.0 Alabama 1.0 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 5.0 88.0 150.0 503.0 3.0 2.0 1.0 1.0 1.0 2.0 2.0 null 2.0 null 1.0 3.0 2.0 null 9.0 99900.0 9.0 1.0 9.0 6.0 1.0 2.0 3.0 2.0 2.0 null 4.0 Alabama 1.0 5.0 2.0 2.0 6.0 1.0 2.0 2.0 1.0 6.0 2.0 170.0 511.0 2.0 2.0 2.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 8.0 1.0 1.0 3.0 1.0 1.0 2.0 2.0 Alabama 1.0 12.0 2.0 3.0 2.0 1.0 8.0 2.0 7.0 99.0 88.0 163.0 503.0 3.0 2.0 1.0 2.0 3.0 1.0 2.0 2.0 2.0 null 1.0 3.0 1.0 888.0 2.0 0.0 1.0 3.0 1.0 12.0 2.0 2.0 2.0 1.0 2.0 2.0 4.0 Alabama 1.0 11.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 134.0 506.0 2.0 1.0 2.0 2.0 3.0 2.0 2.0 1.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 4.0 1.0 2.0 3.0 1.0 1.0 2.0 4.0 Alabama 1.0 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 4.0 88.0 162.0 500.0 4.0 1.0 1.0 2.0 1.0 2.0 2.0 2.0 2.0 null 1.0 3.0 1.0 888.0 2.0 0.0 1.0 3.0 1.0 9.0 2.0 1.0 2.0 1.0 1.0 2.0 3.0 Alabama 1.0 12.0 2.0 3.0 4.0 1.0 1.0 2.0 7.0 null 88.0 null null null 2.0 1.0 2.0 3.0 2.0 2.0 null 7.0 null null null 9.0 null 9.0 99900.0 9.0 9.0 9.0 6.0 1.0 1.0 1.0 1.0 1.0 null 2.0 Alabama 1.0 13.0 2.0 2.0 2.0 1.0 8.0 2.0 7.0 4.0 88.0 140.0 501.0 3.0 2.0 1.0 2.0 4.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 5.0 2.0 2.0 1.0 1.0 1.0 2.0 4.0 Alabama 1.0 10.0 1.0 2.0 4.0 1.0 1.0 2.0 7.0 2.0 88.0 300.0 600.0 4.0 2.0 1.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 10.0 1.0 1.0 1.0 1.0 1.0 2.0 3.0 Alabama 1.0 9.0 1.0 2.0 4.0 1.0 1.0 2.0 8.0 77.0 88.0 260.0 601.0 4.0 1.0 1.0 1.0 1.0 1.0 2.0 2.0 2.0 null 1.0 3.0 2.0 888.0 2.0 0.0 1.0 1.0 1.0 15.0 2.0 2.0 3.0 1.0 1.0 2.0 5.0 Alabama 1.0 8.0 2.0 2.0 4.0 1.0 1.0 2.0 1.0 5.0 88.0 147.0 500.0 3.0 2.0 1.0 1.0 3.0 2.0 2.0 2.0 2.0 null 1.0 3.0 1.0 888.0 2.0 0.0 1.0 3.0 1.0 5.0 1.0 1.0 1.0 1.0 1.0 2.0 2.0 Alabama 1.0 12.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 215.0 601.0 3.0 2.0 1.0 2.0 1.0 2.0 2.0 2.0 2.0 null 1.0 3.0 1.0 230.0 1.0 700.0 1.0 3.0 1.0 8.0 1.0 1.0 1.0 1.0 1.0 2.0 2.0 Alabama 1.0 13.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 164.0 504.0 3.0 2.0 1.0 2.0 3.0 2.0 2.0 2.0 2.0 null 2.0 3.0 1.0 888.0 2.0 0.0 1.0 4.0 1.0 7.0 2.0 1.0 2.0 1.0 1.0 2.0 3.0 Alabama 1.0 9.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 7.0 88.0 150.0 500.0 3.0 2.0 1.0 1.0 3.0 2.0 2.0 2.0 2.0 null 1.0 3.0 2.0 888.0 2.0 0.0 1.0 1.0 1.0 5.0 2.0 1.0 1.0 3.0 1.0 2.0 3.0 Alabama 1.0 7.0 2.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 170.0 504.0 3.0 2.0 2.0 2.0 3.0 2.0 2.0 2.

- **Demographic Dataset**

In [0]:
Demography = brfss2020_df["STATE","STATENAME","AGE","SEXVAR", "MARITAL", "EDUCA", "RENTHOME", "CELLPHONES", "VETERAN", "EMPLOYE","INCOME","CHILDREN","WEIGHT","HEIGHT","BMI","GENHLTH"]
display(Demography)

STATE STATENAME AGE SEXVAR MARITAL EDUCA RENTHOME CELLPHONES VETERAN EMPLOYE INCOME CHILDREN WEIGHT HEIGHT BMI GENHLTH 1.0 Alabama 8.0 2.0 2.0 6.0 1.0 1.0 2.0 4.0 1.0 88.0 106.0 507.0 1.0 2.0 1.0 Alabama 10.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 99.0 88.0 170.0 504.0 3.0 3.0 1.0 Alabama 10.0 2.0 1.0 5.0 1.0 1.0 2.0 7.0 7.0 88.0 7777.0 508.0 null 3.0 1.0 Alabama 13.0 2.0 3.0 4.0 1.0 9.0 2.0 5.0 99.0 88.0 9999.0 9999.0 null 1.0 1.0 Alabama 13.0 2.0 3.0 6.0 2.0 8.0 2.0 7.0 77.0 88.0 126.0 506.0 2.0 2.0 1.0 Alabama 10.0 1.0 4.0 4.0 3.0 1.0 2.0 8.0 5.0 88.0 180.0 509.0 3.0 4.0 1.0 Alabama 12.0 2.0 1.0 4.0 1.0 2.0 2.0 7.0 6.0 88.0 150.0 506.0 2.0 3.0 1.0 Alabama 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 5.0 88.0 150.0 503.0 3.0 4.0 1.0 Alabama 5.0 2.0 2.0 6.0 1.0 2.0 2.0 1.0 6.0 2.0 170.0 511.0 2.0 2.0 1.0 Alabama 12.0 2.0 3.0 2.0 1.0 8.0 2.0 7.0 99.0 88.0 163.0 503.0 3.0 4.0 1.0 Alabama 11.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 134.0 506.0 2.0 4.0 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 4.0 88.0 162.0 500.0 4.0 3.0 1.0 Alabama 12.0 2.0 3.0 4.0 1.0 1.0 2.0 7.0 null 88.0 null null null 2.0 1.0 Alabama 13.0 2.0 2.0 2.0 1.0 8.0 2.0 7.0 4.0 88.0 140.0 501.0 3.0 4.0 1.0 Alabama 10.0 1.0 2.0 4.0 1.0 1.0 2.0 7.0 2.0 88.0 300.0 600.0 4.0 3.0 1.0 Alabama 9.0 1.0 2.0 4.0 1.0 1.0 2.0 8.0 77.0 88.0 260.0 601.0 4.0 5.0 1.0 Alabama 8.0 2.0 2.0 4.0 1.0 1.0 2.0 1.0 5.0 88.0 147.0 500.0 3.0 2.0 1.0 Alabama 12.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 215.0 601.0 3.0 2.0 1.0 Alabama 13.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 164.0 504.0 3.0 3.0 1.0 Alabama 9.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 7.0 88.0 150.0 500.0 3.0 3.0 1.0 Alabama 7.0 2.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 170.0 504.0 3.0 2.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 6.0 88.0 130.0 411.0 3.0 2.0 1.0 Alabama 11.0 1.0 1.0 3.0 1.0 1.0 1.0 7.0 5.0 88.0 162.0 511.0 2.0 3.0 1.0 Alabama 12.0 2.0 3.0 4.0 1.0 1.0 2.0 5.0 2.0 88.0 185.0 506.0 3.0 4.0 1.0 Alabama 10.0 2.0 1.0 5.0 1.0 2.0 2.0 1.0 99.0 88.0 9999.0 500.0 null 2.0 1.0 Alabama 13.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 8.0 88.0 130.0 511.0 1.0 1.0 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 77.0 88.0 112.0 501.0 2.0 3.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 99.0 88.0 9999.0 502.0 null 3.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 5.0 88.0 158.0 502.0 3.0 2.0 1.0 Alabama 6.0 2.0 1.0 4.0 2.0 3.0 2.0 5.0 7.0 1.0 134.0 500.0 3.0 2.0 1.0 Alabama 13.0 1.0 1.0 4.0 1.0 8.0 1.0 7.0 7.0 88.0 160.0 506.0 3.0 4.0 1.0 Alabama 13.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 8.0 88.0 150.0 504.0 3.0 2.0 1.0 Alabama 9.0 2.0 1.0 5.0 1.0 1.0 2.0 8.0 8.0 88.0 170.0 504.0 3.0 5.0 1.0 Alabama 10.0 2.0 5.0 5.0 2.0 1.0 2.0 7.0 2.0 88.0 170.0 411.0 4.0 4.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 1.0 1.0 1.0 8.0 88.0 240.0 601.0 4.0 2.0 1.0 Alabama 8.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 8.0 88.0 145.0 504.0 2.0 2.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 180.0 509.0 3.0 3.0 1.0 Alabama 9.0 2.0 1.0 4.0 1.0 8.0 2.0 5.0 6.0 88.0 200.0 502.0 4.0 3.0 1.0 Alabama 11.0 1.0 1.0 4.0 1.0 2.0 1.0 7.0 8.0 1.0 165.0 507.0 3.0 3.0 1.0 Alabama 13.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 8.0 88.0 190.0 506.0 4.0 4.0 1.0 Alabama 11.0 1.0 1.0 6.0 1.0 1.0 1.0 7.0 8.0 88.0 281.0 506.0 4.0 3.0 1.0 Alabama 9.0 2.0 1.0 5.0 1.0 1.0 2.0 1.0 8.0 88.0 104.0 502.0 2.0 2.0 1.0 Alabama 11.0 2.0 3.0 2.0 2.0 1.0 2.0 8.0 77.0 88.0 220.0 503.0 4.0 3.0 1.0 Alabama 12.0 1.0 1.0 6.0 1.0 2.0 1.0 7.0 7.0 88.0 250.0 601.0 4.0 5.0 1.0 Alabama 13.0 2.0 3.0 3.0 1.0 1.0 2.0 7.0 3.0 88.0 139.0 503.0 2.0 3.0 1.0 Alabama 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 77.0 88.0 119.0 504.0 2.0 4.0 1.0 Alabama 13.0 2.0 3.0 4.0 1.0 1.0 2.0 7.0 4.0 88.0 180.0 508.0 3.0 2.0 1.0 Alabama 11.0 2.0 1.0 6.0 1.0 8.0 2.0 7.0 99.0 88.0 140.0 507.0 2.0 4.0 1.0 Alabama 8.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 8.0 1.0 140.0 506.0 2.0 2.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 2.0 2.0 1.0 8.0 88.0 185.0 508.0 3.0 1.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 4.0 1.0 9999.0 503.0 null 3.0 1.0 Alabama 13.0 2.0 3.0 2.0 1.0 1.0 2.0 7.0 77.0 88.0 160.0 507.0 3.0 3.0 1.0 Alabama 12.0 2.0 1.0 6.0 1

In [0]:
Demography.columns

Out[5]: ['STATE',
 'STATENAME',
 'AGE',
 'SEXVAR',
 'MARITAL',
 'EDUCA',
 'RENTHOME',
 'CELLPHONES',
 'VETERAN',
 'EMPLOYE',
 'INCOME',
 'CHILDREN',
 'WEIGHT',
 'HEIGHT',
 'BMI',
 'GENHLTH']

In [0]:
#print the schema
Demography.printSchema()

root
 |-- STATE: double (nullable = true)
 |-- STATENAME: string (nullable = true)
 |-- AGE: double (nullable = true)
 |-- SEXVAR: double (nullable = true)
 |-- MARITAL: double (nullable = true)
 |-- EDUCA: double (nullable = true)
 |-- RENTHOME: double (nullable = true)
 |-- CELLPHONES: double (nullable = true)
 |-- VETERAN: double (nullable = true)
 |-- EMPLOYE: double (nullable = true)
 |-- INCOME: double (nullable = true)
 |-- CHILDREN: double (nullable = true)
 |-- WEIGHT: double (nullable = true)
 |-- HEIGHT: double (nullable = true)
 |-- BMI: double (nullable = true)
 |-- GENHLTH: double (nullable = true)



In [0]:
#Count the total number of rows and columns
print((Demography.count(), len(Demography.columns)))

(401958, 16)


- **Eliminating the NUll Values**

In [0]:
for col in Demography.columns:
    print(col+":",Demography[Demography[col].isNull()].count())

STATE: 0
STATENAME: 0
AGE: 0
SEXVAR: 0
MARITAL: 12
EDUCA: 12
RENTHOME: 15
CELLPHONES: 116
VETERAN: 1307
EMPLOYE: 2925
INCOME: 7358
CHILDREN: 4767
WEIGHT: 9852
HEIGHT: 10824
BMI: 41357
GENHLTH: 8


In [0]:
#drop the null value
Demography = Demography.na.drop(subset=["STATE","STATENAME","AGE","SEXVAR", "MARITAL", "EDUCA", "RENTHOME", "CELLPHONES", "VETERAN", "EMPLOYE","INCOME","CHILDREN","WEIGHT","HEIGHT","BMI","GENHLTH"])
display(Demography)

STATE STATENAME AGE SEXVAR MARITAL EDUCA RENTHOME CELLPHONES VETERAN EMPLOYE INCOME CHILDREN WEIGHT HEIGHT BMI GENHLTH 1.0 Alabama 8.0 2.0 2.0 6.0 1.0 1.0 2.0 4.0 1.0 88.0 106.0 507.0 1.0 2.0 1.0 Alabama 10.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 99.0 88.0 170.0 504.0 3.0 3.0 1.0 Alabama 13.0 2.0 3.0 6.0 2.0 8.0 2.0 7.0 77.0 88.0 126.0 506.0 2.0 2.0 1.0 Alabama 10.0 1.0 4.0 4.0 3.0 1.0 2.0 8.0 5.0 88.0 180.0 509.0 3.0 4.0 1.0 Alabama 12.0 2.0 1.0 4.0 1.0 2.0 2.0 7.0 6.0 88.0 150.0 506.0 2.0 3.0 1.0 Alabama 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 5.0 88.0 150.0 503.0 3.0 4.0 1.0 Alabama 5.0 2.0 2.0 6.0 1.0 2.0 2.0 1.0 6.0 2.0 170.0 511.0 2.0 2.0 1.0 Alabama 12.0 2.0 3.0 2.0 1.0 8.0 2.0 7.0 99.0 88.0 163.0 503.0 3.0 4.0 1.0 Alabama 11.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 134.0 506.0 2.0 4.0 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 4.0 88.0 162.0 500.0 4.0 3.0 1.0 Alabama 13.0 2.0 2.0 2.0 1.0 8.0 2.0 7.0 4.0 88.0 140.0 501.0 3.0 4.0 1.0 Alabama 10.0 1.0 2.0 4.0 1.0 1.0 2.0 7.0 2.0 88.0 300.0 600.0 4.0 3.0 1.0 Alabama 9.0 1.0 2.0 4.0 1.0 1.0 2.0 8.0 77.0 88.0 260.0 601.0 4.0 5.0 1.0 Alabama 8.0 2.0 2.0 4.0 1.0 1.0 2.0 1.0 5.0 88.0 147.0 500.0 3.0 2.0 1.0 Alabama 12.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 215.0 601.0 3.0 2.0 1.0 Alabama 13.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 164.0 504.0 3.0 3.0 1.0 Alabama 9.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 7.0 88.0 150.0 500.0 3.0 3.0 1.0 Alabama 7.0 2.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 170.0 504.0 3.0 2.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 6.0 88.0 130.0 411.0 3.0 2.0 1.0 Alabama 11.0 1.0 1.0 3.0 1.0 1.0 1.0 7.0 5.0 88.0 162.0 511.0 2.0 3.0 1.0 Alabama 12.0 2.0 3.0 4.0 1.0 1.0 2.0 5.0 2.0 88.0 185.0 506.0 3.0 4.0 1.0 Alabama 13.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 8.0 88.0 130.0 511.0 1.0 1.0 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 77.0 88.0 112.0 501.0 2.0 3.0 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 5.0 88.0 158.0 502.0 3.0 2.0 1.0 Alabama 6.0 2.0 1.0 4.0 2.0 3.0 2.0 5.0 7.0 1.0 134.0 500.0 3.0 2.0 1.0 Alabama 13.0 1.0 1.0 4.0 1.0 8.0 1.0 7.0 7.0 88.0 160.0 506.0 3.0 4.0 1.0 Alabama 13.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 8.0 88.0 150.0 504.0 3.0 2.0 1.0 Alabama 9.0 2.0 1.0 5.0 1.0 1.0 2.0 8.0 8.0 88.0 170.0 504.0 3.0 5.0 1.0 Alabama 10.0 2.0 5.0 5.0 2.0 1.0 2.0 7.0 2.0 88.0 170.0 411.0 4.0 4.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 1.0 1.0 1.0 8.0 88.0 240.0 601.0 4.0 2.0 1.0 Alabama 8.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 8.0 88.0 145.0 504.0 2.0 2.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 180.0 509.0 3.0 3.0 1.0 Alabama 9.0 2.0 1.0 4.0 1.0 8.0 2.0 5.0 6.0 88.0 200.0 502.0 4.0 3.0 1.0 Alabama 11.0 1.0 1.0 4.0 1.0 2.0 1.0 7.0 8.0 1.0 165.0 507.0 3.0 3.0 1.0 Alabama 13.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 8.0 88.0 190.0 506.0 4.0 4.0 1.0 Alabama 11.0 1.0 1.0 6.0 1.0 1.0 1.0 7.0 8.0 88.0 281.0 506.0 4.0 3.0 1.0 Alabama 9.0 2.0 1.0 5.0 1.0 1.0 2.0 1.0 8.0 88.0 104.0 502.0 2.0 2.0 1.0 Alabama 11.0 2.0 3.0 2.0 2.0 1.0 2.0 8.0 77.0 88.0 220.0 503.0 4.0 3.0 1.0 Alabama 12.0 1.0 1.0 6.0 1.0 2.0 1.0 7.0 7.0 88.0 250.0 601.0 4.0 5.0 1.0 Alabama 13.0 2.0 3.0 3.0 1.0 1.0 2.0 7.0 3.0 88.0 139.0 503.0 2.0 3.0 1.0 Alabama 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 77.0 88.0 119.0 504.0 2.0 4.0 1.0 Alabama 13.0 2.0 3.0 4.0 1.0 1.0 2.0 7.0 4.0 88.0 180.0 508.0 3.0 2.0 1.0 Alabama 11.0 2.0 1.0 6.0 1.0 8.0 2.0 7.0 99.0 88.0 140.0 507.0 2.0 4.0 1.0 Alabama 8.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 8.0 1.0 140.0 506.0 2.0 2.0 1.0 Alabama 9.0 1.0 1.0 6.0 1.0 2.0 2.0 1.0 8.0 88.0 185.0 508.0 3.0 1.0 1.0 Alabama 13.0 2.0 3.0 2.0 1.0 1.0 2.0 7.0 77.0 88.0 160.0 507.0 3.0 3.0 1.0 Alabama 12.0 2.0 1.0 6.0 1.0 2.0 2.0 7.0 77.0 88.0 160.0 501.0 4.0 4.0 1.0 Alabama 8.0 1.0 1.0 4.0 3.0 2.0 1.0 8.0 4.0 88.0 230.0 601.0 4.0 4.0 1.0 Alabama 11.0 2.0 3.0 3.0 1.0 1.0 2.0 7.0 5.0 88.0 134.0 508.0 2.0 5.0 1.0 Alabama 10.0 1.0 1.0 6.0 1.0 1.0 1.0 7.0 6.0 88.0 245.0 600.0 4.0 2.0 1.0 Alabama 10.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 6.0 88.0 175.0 510.0 3.0 3.0 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 5.0 88.0 103.0 500.0 2.0 4.0 1.0 Alabama 12.0 1.0 2.0 4.0 1.0 8.0 1.0 7.

- **About the Demographic data**

**How many states in the dataset?**

In [0]:
Demography.select("STATE").distinct().count()

Out[10]: 53

In [0]:
df1 = Demography.groupBy("STATENAME").count()

In [0]:
display(df1.sort("count"))

STATENAME count Guam 2058 Nevada 2280 District of Columbia 3126 Illinois 3270 Alaska 3334 Delaware 3520 Kentucky 3659 South Carolina 3703 North Dakota 4076 Tennessee 4152 Louisiana 4319 California 4354 Wyoming 4410 Oklahoma 4530 Wisconsin 4632 Rhode Island 4671 Arkansas 4695 Puerto Rico 4702 Oregon 4830 Alabama 4973 Pennsylvania 5024 North Carolina 5160 Idaho 5378 West Virginia 5467 New Hampshire 5596 Montana 5852 Vermont 5856 Mississippi 6075 Massachusetts 6124 South Dakota 6231 New Mexico 6485 Michigan 6757 Hawaii 7383 Indiana 7482 Connecticut 7930 Georgia 8072 Missouri 8439 Virginia 8568 Iowa 8692 Arizona 9167 Colorado 9211 New Jersey 9392 Kansas 9592 Texas 9641 Maine 9682 Utah 9880 Florida 10591 Washington 11388 Maryland 12562 New York 12583 Ohio 13190 Nebraska 13601 Minnesota 14247

In [0]:
display(df1.sort("count"))

STATENAME count Guam 2058 Nevada 2280 District of Columbia 3126 Illinois 3270 Alaska 3334 Delaware 3520 Kentucky 3659 South Carolina 3703 North Dakota 4076 Tennessee 4152 Louisiana 4319 California 4354 Wyoming 4410 Oklahoma 4530 Wisconsin 4632 Rhode Island 4671 Arkansas 4695 Puerto Rico 4702 Oregon 4830 Alabama 4973 Pennsylvania 5024 North Carolina 5160 Idaho 5378 West Virginia 5467 New Hampshire 5596 Montana 5852 Vermont 5856 Mississippi 6075 Massachusetts 6124 South Dakota 6231 New Mexico 6485 Michigan 6757 Hawaii 7383 Indiana 7482 Connecticut 7930 Georgia 8072 Missouri 8439 Virginia 8568 Iowa 8692 Arizona 9167 Colorado 9211 New Jersey 9392 Kansas 9592 Texas 9641 Maine 9682 Utah 9880 Florida 10591 Washington 11388 Maryland 12562 New York 12583 Ohio 13190 Nebraska 13601 Minnesota 14247

Output can only be rendered in Databricks

Minnesota had the highest number of respondents with 14,247, followed by Nebraska with 13,601, Ohio with 13,190, New York with 12,583, and Maryland with 12,562. Washington, Florida,  Utah, and Maine had 11,388,10,591, 9,880, and 9,682 respondents respectively.

- **How many respondent with different AGE Group?**

In [0]:
Demography = Demography.withColumn("AgeGroup", Demography["AGE"])

In [0]:
Demography = Demography.withColumn("AgeGroup", 
                                   when(Demography["AgeGroup"] == '1', 'Age 18 to 24')
                                   .when(Demography["AgeGroup"] == '2', 'Age 25 to 29')
                                   .when(Demography["AgeGroup"] == '3', 'Age 30 to 34')
                                   .when(Demography["AgeGroup"] == '4', 'Age 35 to 39')
                                   .when(Demography["AgeGroup"] == '5', 'Age 40 to 44')
                                   .when(Demography["AgeGroup"] == '6', 'Age 45 to 49')
                                   .when(Demography["AgeGroup"] == '7', 'Age 50 to 54')
                                   .when(Demography["AgeGroup"] == '8', 'Age 55 to 59')
                                   .when(Demography["AgeGroup"] == '9', 'Age 60 to 64')
                                   .when(Demography["AgeGroup"] == '10', 'Age 65 to 69')
                                   .when(Demography["AgeGroup"] == '11', 'Age 70 to 74')
                                   .when(Demography["AgeGroup"] == '12', 'Age 75 to 79')
                                   .when(Demography["AgeGroup"] == '13', 'Age 80 or older')
                                   .when(Demography["AgeGroup"] == '14', 'Blank')
                                   .otherwise('AgeGroup'))

In [0]:
df2 = Demography.groupBy("AgeGroup").count()

In [0]:
display(df2.sort("count"))

AgeGroup count Blank 4109 Age 25 to 29 18615 Age 30 to 34 20656 Age 35 to 39 22621 Age 40 to 44 23030 Age 18 to 24 23154 Age 45 to 49 23827 Age 75 to 79 24598 Age 50 to 54 27989 Age 80 or older 29162 Age 55 to 59 32834 Age 70 to 74 34771 Age 60 to 64 37347 Age 65 to 69 37879

In [0]:
display(df2.sort("count"))

AgeGroup count Blank 4109 Age 25 to 29 18615 Age 30 to 34 20656 Age 35 to 39 22621 Age 40 to 44 23030 Age 18 to 24 23154 Age 45 to 49 23827 Age 75 to 79 24598 Age 50 to 54 27989 Age 80 or older 29162 Age 55 to 59 32834 Age 70 to 74 34771 Age 60 to 64 37347 Age 65 to 69 37879

Output can only be rendered in Databricks

The majority survey was from age group 65 to 69, with 37879 respondents. The next highest number of respondents was from age group 60 to 64, with 37347 respondents. The following age groups had the following number of respondents: 70 to 74 (34771), 55 to 59 (32834), 80 or older (29162), 50 to 54 (27989). 4109 respondents did not provide an answer.

- **Number of Male and Female respondents?**

In [0]:
Demography = Demography.withColumn("SEX", Demography["SEXVAR"])

In [0]:
Demography = Demography.withColumn("SEX", 
                                   when(Demography["SEX"] == '1', 'Male')
                                   .when(Demography["SEX"] == '2', 'Female')
                                   .otherwise('SEX'))

In [0]:
df3 = Demography.groupBy("SEX").count()

In [0]:
display(df3.sort("count"))

SEX count Male 172322 Female 188270

In [0]:
display(df3.sort("count"))

SEX count Male 172322 Female 188270

Output can only be rendered in Databricks

A graph summarizing the data would show that there were more female survey respondents (188270) than male respondents (172322). The graph would indicate that there was a difference of 15948 between the number of male and female respondents.

- **Different Types of Marital Statue**

In [0]:
Demography = Demography.withColumn("MARITALSTATUS", Demography["MARITAL"])


In [0]:
Demography = Demography.withColumn("MARITALSTATUS", 
                                   when(Demography["MARITALSTATUS"] == '1', 'Married ')
                                   .when(Demography["MARITALSTATUS"] == '2', 'Divorced')
                                   .when(Demography["MARITALSTATUS"] == '3', 'Widowed ')
                                   .when(Demography["MARITALSTATUS"] == '4', 'Separated')
                                   .when(Demography["MARITALSTATUS"] == '5', 'Never married ')
                                   .when(Demography["MARITALSTATUS"] == '6', 'A member of an unmarried couple ')
                                   .when(Demography["MARITALSTATUS"] == '9', 'Refused')
                                   .when(Demography["MARITALSTATUS"] == 'Blank', 'Not asked or Missing ')
                                   .otherwise('MARITALSTATUS'))

In [0]:
df4 = Demography.groupBy("MARITALSTATUS").count()

In [0]:
display(df4.sort("count"))

MARITALSTATUS count Refused 1985 Separated 7104 A member of an unmarried couple 13675 Widowed 39655 Divorced 47497 Never married 64793 Married 185883

In [0]:
display(df4.sort("count"))

MARITALSTATUS count Refused 1985 Separated 7104 A member of an unmarried couple 13675 Widowed 39655 Divorced 47497 Never married 64793 Married 185883

Output can only be rendered in Databricks

The survey response from marital status showed that the most common response was married, with 185883 people responding in that category. The next most common response was never married, with 64793 people responding in that category. Divorced and widowed had 47497 and 39655 people responding, respectively. A member of an unmarried couple had 13675 people responding, and separated had 7104 people responding. The least common response was refused, with 1985 people responding.

- **Education Level of the Respondents**

In [0]:
Demography = Demography.withColumn("EDULEVEL", Demography["EDUCA"])


In [0]:
Demography = Demography.withColumn("EDULEVEL", 
                                   when(Demography["EDULEVEL"] == '1', 'Never attended school or only kindergarten')
                                   .when(Demography["EDULEVEL"] == '2', 'Grades 1 through 8 (Elementary) ')
                                    .when(Demography["EDULEVEL"] == '3', ' Grades 9 through 11 (Some high school) ')
                                   .when(Demography["EDULEVEL"] == '4', 'Grade 12 or GED (High school graduate) ')
                                    .when(Demography["EDULEVEL"] == '5', 'College 1 year to 3 years (Some college or technical school) ')
                                   .when(Demography["EDULEVEL"] == '6', 'College 4 years or more (College graduate) ')
                                   .when(Demography["EDULEVEL"] == '9', 'Refused ')
                                   .when(Demography["EDULEVEL"] == 'BLank', 'Not asked or Missing')
                                   .otherwise('EDULEVEL'))

In [0]:
df5 = Demography.groupBy("EDULEVEL").count()

In [0]:
display(df5.sort("count"))

EDULEVEL count Never attended school or only kindergarten 362 Refused 853 Grades 1 through 8 (Elementary) 6398 Grades 9 through 11 (Some high school) 15690 Grade 12 or GED (High school graduate) 96054 College 1 year to 3 years (Some college or technical school) 100571 College 4 years or more (College graduate) 140664

In [0]:
display(df5.sort("count"))

EDULEVEL count Never attended school or only kindergarten 362 Refused 853 Grades 1 through 8 (Elementary) 6398 Grades 9 through 11 (Some high school) 15690 Grade 12 or GED (High school graduate) 96054 College 1 year to 3 years (Some college or technical school) 100571 College 4 years or more (College graduate) 140664

Output can only be rendered in Databricks

The survey respondent for the BRFSS 2020 showed the following education levels: College graduates (4+ years of college) at 140664, some college or technical school (1-3 years of college) at 100571, high school graduates (Grade 12 or GED) at 96054, some high school (Grades 9 through 11) at 15690, no high school diploma (Grade 8 or less) at 2409, and other at 719.

- **Relation between Education, Employment, Income and General Health**

In [0]:
Demography = Demography.withColumn("EMPLOYMENT", Demography["EMPLOYE"])

In [0]:
Demography = Demography.withColumn("EMPLOYMENT", 
                                   when(Demography["EMPLOYMENT"] == '1', 'Employed for wages ')
                                   .when(Demography["EMPLOYMENT"] == '2', 'Self-employed ')
                                   .when(Demography["EMPLOYMENT"] == '3', 'Out of work for 1 year or more')
                                   .when(Demography["EMPLOYMENT"] == '4', 'Out of work for less than 1 year ')
                                   .when(Demography["EMPLOYMENT"] == '5', 'A homemaker ')
                                   .when(Demography["EMPLOYMENT"] == '6', 'A student')
                                   .when(Demography["EMPLOYMENT"] == '7', 'Retired')
                                   .when(Demography["EMPLOYMENT"] == '8', 'Unable to work ')
                                   .when(Demography["EMPLOYMENT"] == '9', 'Refused')
                                   .when(Demography["EMPLOYMENT"] == 'Blank', 'Not asked or Missing ')
                                   .otherwise('EMPLOYMENT'))

In [0]:
Demography = Demography.withColumn("INCOMELEVEL", Demography["INCOME"])

In [0]:
Demography = Demography.withColumn("INCOMELEVEL", 
                                   when(Demography["INCOMELEVEL"] == '1', 'Less than $10,000')
                                   .when(Demography["INCOMELEVEL"] == '2', 'Less than $15,000')
                                   .when(Demography["INCOMELEVEL"] == '3', 'Less than $20,000')
                                   .when(Demography["INCOMELEVEL"] == '4', 'Less than $25,000')
                                   .when(Demography["INCOMELEVEL"] == '5', 'Less than $35,000 ')
                                   .when(Demography["INCOMELEVEL"] == '6', 'Less than $50,000')
                                   .when(Demography["INCOMELEVEL"] == '7', 'Less than $75,000')
                                   .when(Demography["INCOMELEVEL"] == '8', '$75,000 or more ')
                                   .when(Demography["INCOMELEVEL"] == '77', 'Don’t know/Not sure ')
                                   .when(Demography["INCOMELEVEL"] == '99', 'Refused')
                                   .when(Demography["INCOMELEVEL"] == 'Blank', 'Not asked or Missing ')
                                   .otherwise('INCOMELEVEL'))

In [0]:
Demography = Demography.withColumn("GENHLTHSTAT", Demography["GENHLTH"])

In [0]:
Demography = Demography.withColumn("GENHLTHSTAT", 
                                   when(Demography["GENHLTHSTAT"] == '1', 'Excellent ')
                                   .when(Demography["GENHLTHSTAT"] == '2', 'Very good')
                                   .when(Demography["GENHLTHSTAT"] == '3', 'Good ')
                                   .when(Demography["GENHLTHSTAT"] == '4', 'Fair ')
                                   .when(Demography["GENHLTHSTAT"] == '5', 'Poor')
                                   .when(Demography["GENHLTHSTAT"] == '7', 'Don’t know/Not Sure ')
                                   .when(Demography["GENHLTHSTAT"] == '9', 'Refused ')
                                   .when(Demography["GENHLTHSTAT"] == 'Blank', 'Not asked or Missing ')
                                   .otherwise('GENHLTHSTAT'))

In [0]:
df5 = Demography.groupBy("AGEGROUP").count()
display(df5)

AGEGROUP count Age 45 to 49 23827 Age 25 to 29 18615 Age 70 to 74 34771 Blank 4109 Age 55 to 59 32834 Age 18 to 24 23154 Age 60 to 64 37347 Age 50 to 54 27989 Age 35 to 39 22621 Age 30 to 34 20656 Age 80 or older 29162 Age 40 to 44 23030 Age 75 to 79 24598 Age 65 to 69 37879

In [0]:
multiple = Demography\
.groupby(["AGEGROUP","EDULEVEL","EMPLOYMENT","INCOMELEVEL","GENHLTHSTAT"])\
.agg({'AGEGROUP':'count','EDULEVEL':'count','EMPLOYMENT':'count','INCOMELEVEL':'count','GENHLTHSTAT':'count'},)

multiple

Out[41]: DataFrame[AGEGROUP: string, EDULEVEL: string, EMPLOYMENT: string, INCOMELEVEL: string, GENHLTHSTAT: string, count(INCOMELEVEL): bigint, count(EMPLOYMENT): bigint, count(AGEGROUP): bigint, count(EDULEVEL): bigint, count(GENHLTHSTAT): bigint]

In [0]:
display(multiple)

AGEGROUP EDULEVEL EMPLOYMENT INCOMELEVEL GENHLTHSTAT count(INCOMELEVEL) count(EMPLOYMENT) count(AGEGROUP) count(EDULEVEL) count(GENHLTHSTAT) Age 75 to 79 Grade 12 or GED (High school graduate) A homemaker Don’t know/Not sure Fair 12 12 12 12 12 Age 80 or older Grade 12 or GED (High school graduate) Unable to work Less than $20,000 Poor 18 18 18 18 18 Age 45 to 49 College 4 years or more (College graduate) Employed for wages Refused Very good 197 197 197 197 197 Age 80 or older College 4 years or more (College graduate) Unable to work Less than $25,000 Good 4 4 4 4 4 Age 35 to 39 College 1 year to 3 years (Some college or technical school) Employed for wages Less than $50,000 Very good 198 198 198 198 198 Age 35 to 39 Grade 12 or GED (High school graduate) A homemaker Less than $10,000 Excellent 6 6 6 6 6 Age 45 to 49 College 4 years or more (College graduate) Employed for wages Less than $50,000 Good 149 149 149 149 149 Age 50 to 54 Grades 9 through 11 (Some high school) Self-employed Don’t know/Not sure Very good 3 3 3 3 3 Age 50 to 54 College 1 year to 3 years (Some college or technical school) Out of work for less than 1 year Less than $50,000 Good 24 24 24 24 24 Age 55 to 59 Grade 12 or GED (High school graduate) Employed for wages $75,000 or more Poor 9 9 9 9 9 Age 50 to 54 Grades 9 through 11 (Some high school) A homemaker Less than $20,000 Good 3 3 3 3 3 Age 75 to 79 Grade 12 or GED (High school graduate) A homemaker Less than $75,000 Fair 5 5 5 5 5 Age 45 to 49 College 1 year to 3 years (Some college or technical school) Employed for wages Refused Very good 63 63 63 63 63 Blank College 4 years or more (College graduate) Self-employed Don’t know/Not sure Very good 2 2 2 2 2 Age 80 or older Grade 12 or GED (High school graduate) Employed for wages $75,000 or more Poor 2 2 2 2 2 Age 50 to 54 College 4 years or more (College graduate) Retired Less than $75,000 Very good 20 20 20 20 20 Age 45 to 49 College 1 year to 3 years (Some college or technical school) Unable to work Less than $25,000 Poor 19 19 19 19 19 Age 45 to 49 College 4 years or more (College graduate) Out of work for 1 year or more Less than $20,000 Excellent 2 2 2 2 2 Age 55 to 59 College 1 year to 3 years (Some college or technical school) Out of work for less than 1 year Less than $50,000 Good 22 22 22 22 22 Age 70 to 74 College 1 year to 3 years (Some college or technical school) Retired Less than $15,000 Poor 25 25 25 25 25 Age 55 to 59 Grade 12 or GED (High school graduate) A homemaker Don’t know/Not sure Good 19 19 19 19 19 Age 35 to 39 College 4 years or more (College graduate) Out of work for less than 1 year Less than $25,000 Good 7 7 7 7 7 Age 70 to 74 College 1 year to 3 years (Some college or technical school) Self-employed Less than $50,000 Good 32 32 32 32 32 Age 45 to 49 Grades 9 through 11 (Some high school) A homemaker Less than $50,000 Good 6 6 6 6 6 Age 35 to 39 College 1 year to 3 years (Some college or technical school) A student Less than $10,000 Very good 7 7 7 7 7 Age 60 to 64 Grade 12 or GED (High school graduate) Self-employed Less than $20,000 Fair 10 10 10 10 10 Age 80 or older College 4 years or more (College graduate) Retired Less than $20,000 Excellent 24 24 24 24 24 Age 65 to 69 Grades 1 through 8 (Elementary) Refused Less than $15,000 Very good 1 1 1 1 1 Age 50 to 54 Grades 1 through 8 (Elementary) Unable to work Less than $35,000 Fair 3 3 3 3 3 Age 18 to 24 College 1 year to 3 years (Some college or technical school) Refused Less than $25,000 Good 2 2 2 2 2 Age 60 to 64 College 1 year to 3 years (Some college or technical school) Refused Don’t know/Not sure Don’t know/Not Sure 1 1 1 1 1 Age 35 to 39 Grades 1 through 8 (Elementary) Employed for wages Less than $50,000 Fair 2 2 2 2 2 Age 35 to 39 Grades 9 through 11 (Some high school) Self-employed Less than $50,000 Excellent 5 5 5 5 5 Age 35 to 39 Grade 12 or GED (High school graduate) Unable to work Less than $15,000 Good 17 17 17 17 17 Age 35 to 39 Grades 9 through 11 (Some 

In [0]:
display(multiple)

AGEGROUP EDULEVEL EMPLOYMENT INCOMELEVEL GENHLTHSTAT count(INCOMELEVEL) count(EMPLOYMENT) count(AGEGROUP) count(EDULEVEL) count(GENHLTHSTAT) Age 75 to 79 Grade 12 or GED (High school graduate) A homemaker Don’t know/Not sure Fair 12 12 12 12 12 Age 80 or older Grade 12 or GED (High school graduate) Unable to work Less than $20,000 Poor 18 18 18 18 18 Age 45 to 49 College 4 years or more (College graduate) Employed for wages Refused Very good 197 197 197 197 197 Age 80 or older College 4 years or more (College graduate) Unable to work Less than $25,000 Good 4 4 4 4 4 Age 35 to 39 College 1 year to 3 years (Some college or technical school) Employed for wages Less than $50,000 Very good 198 198 198 198 198 Age 35 to 39 Grade 12 or GED (High school graduate) A homemaker Less than $10,000 Excellent 6 6 6 6 6 Age 45 to 49 College 4 years or more (College graduate) Employed for wages Less than $50,000 Good 149 149 149 149 149 Age 50 to 54 Grades 9 through 11 (Some high school) Self-employed Don’t know/Not sure Very good 3 3 3 3 3 Age 50 to 54 College 1 year to 3 years (Some college or technical school) Out of work for less than 1 year Less than $50,000 Good 24 24 24 24 24 Age 55 to 59 Grade 12 or GED (High school graduate) Employed for wages $75,000 or more Poor 9 9 9 9 9 Age 50 to 54 Grades 9 through 11 (Some high school) A homemaker Less than $20,000 Good 3 3 3 3 3 Age 75 to 79 Grade 12 or GED (High school graduate) A homemaker Less than $75,000 Fair 5 5 5 5 5 Age 45 to 49 College 1 year to 3 years (Some college or technical school) Employed for wages Refused Very good 63 63 63 63 63 Blank College 4 years or more (College graduate) Self-employed Don’t know/Not sure Very good 2 2 2 2 2 Age 80 or older Grade 12 or GED (High school graduate) Employed for wages $75,000 or more Poor 2 2 2 2 2 Age 50 to 54 College 4 years or more (College graduate) Retired Less than $75,000 Very good 20 20 20 20 20 Age 45 to 49 College 1 year to 3 years (Some college or technical school) Unable to work Less than $25,000 Poor 19 19 19 19 19 Age 45 to 49 College 4 years or more (College graduate) Out of work for 1 year or more Less than $20,000 Excellent 2 2 2 2 2 Age 55 to 59 College 1 year to 3 years (Some college or technical school) Out of work for less than 1 year Less than $50,000 Good 22 22 22 22 22 Age 70 to 74 College 1 year to 3 years (Some college or technical school) Retired Less than $15,000 Poor 25 25 25 25 25 Age 55 to 59 Grade 12 or GED (High school graduate) A homemaker Don’t know/Not sure Good 19 19 19 19 19 Age 35 to 39 College 4 years or more (College graduate) Out of work for less than 1 year Less than $25,000 Good 7 7 7 7 7 Age 70 to 74 College 1 year to 3 years (Some college or technical school) Self-employed Less than $50,000 Good 32 32 32 32 32 Age 45 to 49 Grades 9 through 11 (Some high school) A homemaker Less than $50,000 Good 6 6 6 6 6 Age 35 to 39 College 1 year to 3 years (Some college or technical school) A student Less than $10,000 Very good 7 7 7 7 7 Age 60 to 64 Grade 12 or GED (High school graduate) Self-employed Less than $20,000 Fair 10 10 10 10 10 Age 80 or older College 4 years or more (College graduate) Retired Less than $20,000 Excellent 24 24 24 24 24 Age 65 to 69 Grades 1 through 8 (Elementary) Refused Less than $15,000 Very good 1 1 1 1 1 Age 50 to 54 Grades 1 through 8 (Elementary) Unable to work Less than $35,000 Fair 3 3 3 3 3 Age 18 to 24 College 1 year to 3 years (Some college or technical school) Refused Less than $25,000 Good 2 2 2 2 2 Age 60 to 64 College 1 year to 3 years (Some college or technical school) Refused Don’t know/Not sure Don’t know/Not Sure 1 1 1 1 1 Age 35 to 39 Grades 1 through 8 (Elementary) Employed for wages Less than $50,000 Fair 2 2 2 2 2 Age 35 to 39 Grades 9 through 11 (Some high school) Self-employed Less than $50,000 Excellent 5 5 5 5 5 Age 35 to 39 Grade 12 or GED (High school graduate) Unable to work Less than $15,000 Good 17 17 17 17 17 Age 35 to 39 Grades 9 through 11 (Some 

Output can only be rendered in Databricks

In [0]:
multiple2 = Demography\
.groupby(["EDULEVEL","EMPLOYMENT","INCOME","GENHLTH","GENHLTHSTAT"])\
.agg({'GENHLTH':'count','EMPLOYMENT':'count','INCOME':'count'})

multiple2

Out[49]: DataFrame[EDULEVEL: string, EMPLOYMENT: string, INCOME: double, GENHLTH: double, GENHLTHSTAT: string, count(INCOME): bigint, count(GENHLTH): bigint, count(EMPLOYMENT): bigint]

In [0]:
display(multiple2)

EDULEVEL EMPLOYMENT INCOME GENHLTH GENHLTHSTAT count(INCOME) count(GENHLTH) count(EMPLOYMENT) Grades 9 through 11 (Some high school) Out of work for 1 year or more 2.0 3.0 Good 23 23 23 Grade 12 or GED (High school graduate) A student 77.0 1.0 Excellent 218 218 218 Refused A homemaker 2.0 2.0 Very good 2 2 2 Refused Self-employed 7.0 7.0 Don’t know/Not Sure 1 1 1 Grades 1 through 8 (Elementary) A student 1.0 3.0 Good 1 1 1 College 4 years or more (College graduate) Employed for wages 99.0 2.0 Very good 2024 2024 2024 Grade 12 or GED (High school graduate) Retired 7.0 1.0 Excellent 341 341 341 College 1 year to 3 years (Some college or technical school) Out of work for 1 year or more 1.0 1.0 Excellent 46 46 46 College 4 years or more (College graduate) A student 77.0 4.0 Fair 21 21 21 Grades 9 through 11 (Some high school) Out of work for less than 1 year 2.0 1.0 Excellent 6 6 6 College 1 year to 3 years (Some college or technical school) A homemaker 1.0 5.0 Poor 9 9 9 Grades 9 through 11 (Some high school) Unable to work 8.0 5.0 Poor 8 8 8 Grades 1 through 8 (Elementary) Self-employed 7.0 5.0 Poor 2 2 2 College 1 year to 3 years (Some college or technical school) A homemaker 2.0 7.0 Don’t know/Not Sure 1 1 1 Grades 1 through 8 (Elementary) Unable to work 77.0 5.0 Poor 85 85 85 Refused Retired 6.0 3.0 Good 5 5 5 Never attended school or only kindergarten Retired 8.0 3.0 Good 2 2 2 College 1 year to 3 years (Some college or technical school) Refused 99.0 9.0 Refused 1 1 1 Never attended school or only kindergarten Out of work for less than 1 year 3.0 4.0 Fair 1 1 1 Refused Out of work for less than 1 year 2.0 5.0 Poor 1 1 1 College 4 years or more (College graduate) Employed for wages 5.0 4.0 Fair 155 155 155 Grade 12 or GED (High school graduate) Unable to work 99.0 1.0 Excellent 26 26 26 Grades 9 through 11 (Some high school) Out of work for 1 year or more 1.0 2.0 Very good 31 31 31 Grades 9 through 11 (Some high school) Self-employed 4.0 1.0 Excellent 26 26 26 College 1 year to 3 years (Some college or technical school) Self-employed 2.0 5.0 Poor 6 6 6 Never attended school or only kindergarten Employed for wages 5.0 3.0 Good 5 5 5 Never attended school or only kindergarten Employed for wages 99.0 3.0 Good 6 6 6 Refused Out of work for less than 1 year 7.0 3.0 Good 2 2 2 College 1 year to 3 years (Some college or technical school) A student 99.0 2.0 Very good 68 68 68 Never attended school or only kindergarten A homemaker 2.0 5.0 Poor 1 1 1 College 1 year to 3 years (Some college or technical school) A homemaker 8.0 4.0 Fair 62 62 62 College 1 year to 3 years (Some college or technical school) A student 8.0 2.0 Very good 420 420 420 College 4 years or more (College graduate) Out of work for less than 1 year 4.0 2.0 Very good 125 125 125 Grade 12 or GED (High school graduate) Unable to work 6.0 5.0 Poor 95 95 95 Grades 9 through 11 (Some high school) Self-employed 3.0 1.0 Excellent 29 29 29 Grade 12 or GED (High school graduate) Unable to work 7.0 5.0 Poor 60 60 60 College 4 years or more (College graduate) Out of work for less than 1 year 2.0 2.0 Very good 43 43 43 College 4 years or more (College graduate) Self-employed 8.0 5.0 Poor 35 35 35 Grades 1 through 8 (Elementary) Unable to work 8.0 3.0 Good 2 2 2 Refused A homemaker 77.0 2.0 Very good 1 1 1 Never attended school or only kindergarten Employed for wages 8.0 2.0 Very good 4 4 4 Grades 1 through 8 (Elementary) Unable to work 6.0 7.0 Don’t know/Not Sure 1 1 1 College 1 year to 3 years (Some college or technical school) Unable to work 7.0 4.0 Fair 121 121 121 College 1 year to 3 years (Some college or technical school) A student 5.0 2.0 Very good 112 112 112 Grades 9 through 11 (Some high school) Out of work for less than 1 year 2.0 4.0 Fair 13 13 13 College 1 year to 3 years (Some college or technical school) A homemaker 77.0 7.0 Don’t know/Not Sure 3 3 3 Never attended school or only kindergarten Out of work for 1 year or more 1.0 4.0 Fair 2 2 2 College 4 years or more

Output can only be rendered in Databricks

In [0]:
display(Demography)

STATE STATENAME AGE SEXVAR MARITAL EDUCA RENTHOME CELLPHONES VETERAN EMPLOYE INCOME CHILDREN WEIGHT HEIGHT BMI GENHLTH AgeGroup SEX MARITALSTATUS EDULEVEL EMPLOYMENT INCOMELEVEL GENHLTHSTAT 1.0 Alabama 8.0 2.0 2.0 6.0 1.0 1.0 2.0 4.0 1.0 88.0 106.0 507.0 1.0 2.0 Age 55 to 59 Female Divorced College 4 years or more (College graduate) Out of work for less than 1 year Less than $10,000 Very good 1.0 Alabama 10.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 99.0 88.0 170.0 504.0 3.0 3.0 Age 65 to 69 Female Widowed College 4 years or more (College graduate) Retired Refused Good 1.0 Alabama 13.0 2.0 3.0 6.0 2.0 8.0 2.0 7.0 77.0 88.0 126.0 506.0 2.0 2.0 Age 80 or older Female Widowed College 4 years or more (College graduate) Retired Don’t know/Not sure Very good 1.0 Alabama 10.0 1.0 4.0 4.0 3.0 1.0 2.0 8.0 5.0 88.0 180.0 509.0 3.0 4.0 Age 65 to 69 Male Separated Grade 12 or GED (High school graduate) Unable to work Less than $35,000 Fair 1.0 Alabama 12.0 2.0 1.0 4.0 1.0 2.0 2.0 7.0 6.0 88.0 150.0 506.0 2.0 3.0 Age 75 to 79 Female Married Grade 12 or GED (High school graduate) Retired Less than $50,000 Good 1.0 Alabama 10.0 2.0 1.0 4.0 1.0 1.0 2.0 7.0 5.0 88.0 150.0 503.0 3.0 4.0 Age 65 to 69 Female Married Grade 12 or GED (High school graduate) Retired Less than $35,000 Fair 1.0 Alabama 5.0 2.0 2.0 6.0 1.0 2.0 2.0 1.0 6.0 2.0 170.0 511.0 2.0 2.0 Age 40 to 44 Female Divorced College 4 years or more (College graduate) Employed for wages Less than $50,000 Very good 1.0 Alabama 12.0 2.0 3.0 2.0 1.0 8.0 2.0 7.0 99.0 88.0 163.0 503.0 3.0 4.0 Age 75 to 79 Female Widowed Grades 1 through 8 (Elementary) Retired Refused Fair 1.0 Alabama 11.0 2.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 134.0 506.0 2.0 4.0 Age 70 to 74 Female Married College 4 years or more (College graduate) Retired Less than $75,000 Fair 1.0 Alabama 13.0 2.0 3.0 5.0 1.0 1.0 2.0 7.0 4.0 88.0 162.0 500.0 4.0 3.0 Age 80 or older Female Widowed College 1 year to 3 years (Some college or technical school) Retired Less than $25,000 Good 1.0 Alabama 13.0 2.0 2.0 2.0 1.0 8.0 2.0 7.0 4.0 88.0 140.0 501.0 3.0 4.0 Age 80 or older Female Divorced Grades 1 through 8 (Elementary) Retired Less than $25,000 Fair 1.0 Alabama 10.0 1.0 2.0 4.0 1.0 1.0 2.0 7.0 2.0 88.0 300.0 600.0 4.0 3.0 Age 65 to 69 Male Divorced Grade 12 or GED (High school graduate) Retired Less than $15,000 Good 1.0 Alabama 9.0 1.0 2.0 4.0 1.0 1.0 2.0 8.0 77.0 88.0 260.0 601.0 4.0 5.0 Age 60 to 64 Male Divorced Grade 12 or GED (High school graduate) Unable to work Don’t know/Not sure Poor 1.0 Alabama 8.0 2.0 2.0 4.0 1.0 1.0 2.0 1.0 5.0 88.0 147.0 500.0 3.0 2.0 Age 55 to 59 Female Divorced Grade 12 or GED (High school graduate) Employed for wages Less than $35,000 Very good 1.0 Alabama 12.0 1.0 1.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 215.0 601.0 3.0 2.0 Age 75 to 79 Male Married College 4 years or more (College graduate) Retired Less than $75,000 Very good 1.0 Alabama 13.0 2.0 3.0 6.0 1.0 1.0 2.0 7.0 7.0 88.0 164.0 504.0 3.0 3.0 Age 80 or older Female Widowed College 4 years or more (College graduate) Retired Less than $75,000 Good 1.0 Alabama 9.0 2.0 1.0 4.0 1.0 1.0 2.0 5.0 7.0 88.0 150.0 500.0 3.0 3.0 Age 60 to 64 Female Married Grade 12 or GED (High school graduate) A homemaker Less than $75,000 Good 1.0 Alabama 7.0 2.0 1.0 6.0 1.0 1.0 2.0 1.0 8.0 88.0 170.0 504.0 3.0 2.0 Age 50 to 54 Female Married College 4 years or more (College graduate) Employed for wages $75,000 or more Very good 1.0 Alabama 11.0 2.0 1.0 5.0 1.0 2.0 2.0 7.0 6.0 88.0 130.0 411.0 3.0 2.0 Age 70 to 74 Female Married College 1 year to 3 years (Some college or technical school) Retired Less than $50,000 Very good 1.0 Alabama 11.0 1.0 1.0 3.0 1.0 1.0 1.0 7.0 5.0 88.0 162.0 511.0 2.0 3.0 Age 70 to 74 Male Married Grades 9 through 11 (Some high school) Retired Less than $35,000 Good 1.0 Alabama 12.0 2.0 3.0 4.0 1.0 1.0 2.0 5.0 2.0 88.0 185.0 506.0 3.0 4.0 Age 75 to 79 Female Widowed Grade 12 or GED (High school graduate) A homemaker Less than $15,000 Fair 1.0 Alabama 13.0 1.0 1.0 6.0 1.0

Output can only be rendered in Databricks